docs/articles_en/documentation/openvino-extensibility/custom-openvino-operations.rst

```diff
@@ -7,30 +7,64 @@ Custom OpenVINO Operations
+      :sync: py
 
-Follow the steps below to add a custom operation:
+      .. doxygensnippet:: docs/articles_en/assets/snippets/ov_custom_op.py
+         :language: python
+         :fragment: [op:common_include]
 
-1. Add the ``OPENVINO_OP`` macro which defines a ``NodeTypeInfo`` object that identifies the type of the operation to the graph users and helps with dynamic type resolution. The type info of an operation currently consists of a string operation identifier and a string for operation version.
+      Follow the steps below to add a simple custom operation:
 
-2. Implement default constructor and constructors that optionally take the operation inputs and attributes as parameters.
+      1. Define the ``__init__`` method to initialize the class with inputs and attributes.
 
-3. Override the shape inference method ``validate_and_infer_types``. This method is called multiple times during graph manipulations to determine the shapes and element types of the operations outputs. To access the input shapes and input element types, use the ``get_input_partial_shape()`` and ``get_input_element_type()`` methods of ``ov::Node``. Set the inferred shape and element type of the output using ``set_output_type``.
+      2. Override the shape inference method ``validate_and_infer_types``. This method is called multiple times during graph manipulations to determine the shapes and element types of the operations outputs. To access the input shapes and input element types, use the ``get_input_partial_shape()`` and ``get_input_element_type()`` methods of ``openvino.Node``. Set the inferred shape and element type of the output using ``set_output_type``.
 
-4. Override the ``clone_with_new_inputs`` method, which enables graph manipulation routines to create copies of this operation and connect it to different nodes during optimization.
+      3. Override the ``visit_attributes`` method, which enables serialization and deserialization of operation attributes. An ``AttributeVisitor`` is passed to the method, and the implementation is expected to walk over all the attributes in the op using the type-aware ``on_attribute`` helper. Helpers are already implemented for standard types like ``int``, ``float``, ``bool``, ``vector``, and for existing OpenVINO defined types.
 
-5. Override the ``visit_attributes`` method, which enables serialization and deserialization of operation attributes. An ``AttributeVisitor`` is passed to the method, and the implementation is expected to walk over all the attributes in the op using the type-aware ``on_attribute`` helper. Helpers are already implemented for standard C++ types like ``int64_t``, ``float``, ``bool``, ``vector``, and for existing OpenVINO defined types.
+      There are some methods that can also be overridden if needed:
+
+      4. Override the ``clone_with_new_inputs`` method, which enables graph manipulation routines to create copies of this operation and connect it to different nodes during optimization.
```

Comment:
The description doesn't provide any reason why this function is optional. If I don't implement it, what does happen?

Reply:
the default ones will be called.
but overall I removed this phrase

Reply:
So, now `clone_with_new_inputs` is mandatory according to the documentation? That was one of the things we were trying to fix for Python (this is one of the "Python Exclusives" in this case). And now we are not emphasizing it.

Reply:
I've edited the sentence and added remark that it is optional